### PR TITLE
docs: add stream processing examples

### PR DESCRIPTION
**Pull request by derekjn**
Fills in some stream processing example TODOs.

### CODE REVIEW

File: docs-md/concepts/stream-processing.md [hunk 1]
```diff
@@ -53,14 +53,91 @@ CREATE STREAM title_cased_songs AS
...
-TODO: Deriving a new table from an existing stream
---------------------------------------------------
+Given the following table:
+
+```sql
+CREATE TABLE products (product_name VARCHAR, cost DOUBLE)
+    WITH (kafka_topic='products', partitions=2, value_format='json');
+```
+
+You can create a stream that will capture all writes made to the table:
```
Comment (JimGalasyn) [inline docs-md/concepts/stream-processing.md]:
```suggestion
You can create a stream that captures all writes made to the table:
```

File: docs-md/concepts/stream-processing.md [hunk 2]
```diff
@@ -53,14 +53,91 @@ CREATE STREAM title_cased_songs AS
...
+
+```sql
+CREATE STREAM products_changelog AS
+    SELECT * FROM products EMIT CHANGES;
+```
+
+The `products_changelog` stream will emit a new row any time a row in the `products` table is
```
Comment (JimGalasyn) [inline docs-md/concepts/stream-processing.md]:
```suggestion
The `products_changelog` stream emits a new row any time a row in the `products` table is
```

File: docs-md/concepts/stream-processing.md [hunk 3]
```diff
@@ -53,14 +53,91 @@ CREATE STREAM title_cased_songs AS
...
+
+```sql
+CREATE TABLE order_metrics AS
+    SELECT p.product_name, COUNT(*) AS count, SUM(p.cost) AS revenue
+    FROM orders o JOIN products p ON p.product_name = o.product_name
+    GROUP BY p.product_name EMIT CHANGES;
+```
 
-TODO: Deriving a new table from an existing table
+This aggregate table will keep track of the total number of orders of each product, along with the total amount of revenue
```
Comment (JimGalasyn) [inline docs-md/concepts/stream-processing.md]:
```suggestion
This aggregate table keeps track of the total number of orders of each product, along with the total amount of revenue
```

File: docs-md/concepts/stream-processing.md [hunk 4]
```diff
@@ -53,14 +53,91 @@ CREATE STREAM title_cased_songs AS
...
+```sql
+CREATE STREAM impressions (user VARCHAR, impression_id BIGINT, url VARCHAR)
+    WITH (kafka_topic='impressions', partitions=1, value_format='json', key='user');
+
+CREATE STREAM clicks (user VARCHAR)
+    WITH (kafka_topic='clicks', partitions=1, value_format='json', key='user');
+```
+
+You can create a derived stream that joins the `impressions` and `clicks` streams to output
+rows indicating that a given impression has been clicked on within one minute of the initial ad impression:
```
Comment (JimGalasyn) [inline docs-md/concepts/stream-processing.md]:
```suggestion
rows indicating that a given impression has been clicked within one minute of the initial ad impression:
```

File: docs-md/concepts/stream-processing.md [hunk 5]
```diff
@@ -53,14 +53,91 @@ CREATE STREAM title_cased_songs AS
...
+
+```sql
+CREATE STREAM clicked_impressions AS
+    SELECT * FROM impressions i JOIN clicks c WITHIN 1 minute ON i.user = c.user EMIT CHANGES;
+```
+
+Any time an `impressions` row is received, followed within one minute by a `clicks` row having the same `user`, 
+a row will be emitted into the `clicked_impressions` stream.
```
Comment (JimGalasyn) [inline docs-md/concepts/stream-processing.md]:
```suggestion
a row is emitted into the `clicked_impressions` stream.
```

**Comment by derekjn** [comment]
Thanks @JimGalasyn! I wrapped all non-code lines at 80 chars so this should be good to merge when you're ready.

**Pull request merged**